bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/Resource.java

```diff
@@ -165,29 +162,29 @@ public Device getDevice() {
 
 void ignoreNonDisposed() {
 	if (tracker != null) {
-		tracker.ignoreMe = true;
+		tracker.reporting.set(false);
 	}
 }
 
 void init() {
 	if (device.tracking) device.new_Object(this);
+	if (tracker != null && tracker.reporting.compareAndSet(false, true)) {
+		cleaner.register(this, tracker);
```

Comment:
Save the return value here to new field cleanable.

```suggestion
		cleanable = cleaner.register(this, tracker);
```